ECMAScript/react/src/hooks/useQuilttConnector.ts

```diff
@@ -1,39 +1,97 @@
 'use client'
 
-import { useEffect } from 'react'
+import { useCallback, useEffect, useState } from 'react'
 import { useQuilttSession } from './useQuilttSession'
+import { useScript } from './useScript'
+import { ConnectorSDK, ConnectorSDKConnector, ConnectorSDKConnectorOptions } from '@quiltt/core'
 
 const QUILTT_CDN_BASE = process.env.QUILTT_CDN_BASE || 'https://cdn.quiltt.io'
 
-// Script Element Singleton
-let scriptElement: HTMLScriptElement
+declare const Quiltt: ConnectorSDK
 
-export const useQuilttConnector = () => {
+export const useQuilttConnector = (
+  connectorId?: string,
+  options?: ConnectorSDKConnectorOptions
+) => {
+  const status = useScript(`${QUILTT_CDN_BASE}/v1/connector.js`)
   const { session } = useQuilttSession()
+  const [connector, setConnector] = useState<ConnectorSDKConnector>()
+  const [isOpening, setIsOpening] = useState<boolean>(false)
 
-  // Create Script Element
+  // Set Session
   useEffect(() => {
-    if (scriptElement) return
+    if (typeof Quiltt === 'undefined') return
 
-    scriptElement = document.createElement('script')
-    scriptElement.src = `${QUILTT_CDN_BASE}/v1/connector.js`
+    Quiltt.authenticate(session?.token)
```

Comment:
@sirwolfgang should this run even if there is no token in the session?

Reply:
Define "no" token, `undefined` shouldn't do anything, `null` should clear.

Reply:
Ok got it. If I understand correctly, The session has a token or there is no session, so null isn't expected.